onnxruntime/core/framework/op_node_proto_helper.cc

```diff
@@ -90,34 +90,38 @@ inline constexpr int ArrayTypeToAttributeType<std::string>() {
+  template <>                                                                                                \
+  template <>                                                                                                \
+  Status OpNodeProtoHelper<IMPL_T>::GetAttrs<T>(                                                             \
+      const std::string& name, std::vector<T>& values) const {                                               \
+    const AttributeProto* attr = TryGetAttribute(name);                                                      \
+    if (!attr) {                                                                                             \
+      return Status(ONNXRUNTIME, FAIL, "No attribute with this name is defined.");                           \
+    }                                                                                                        \
+    values.reserve(attr->list##_size());                                                                     \
+    for (int i = 0; i < attr->list##_size(); ++i) {                                                          \
+      values.push_back(static_cast<T>(attr->list(i)));                                                       \
+    }                                                                                                        \
+    return Status::OK();                                                                                     \
+  }                                                                                                          \
+  template <>                                                                                                \
+  template <>                                                                                                \
+  Status OpNodeProtoHelper<IMPL_T>::GetAttrs<T>(                                                             \
+      const std::string& name, gsl::span<T> values) const {                                                  \
+    const AttributeProto* attr = TryGetAttribute(name);                                                      \
+    if (!attr) {                                                                                             \
+      return Status(ONNXRUNTIME, FAIL, "No attribute with this name is defined.");                           \
+    }                                                                                                        \
+    if (values.size() != static_cast<size_t>(attr->list##_size())) {                                         \
+      std::ostringstream oss;                                                                                \
+      oss << "GetAttrs failed. Expect values.size()=" << (attr->list##_size()) << ", got " << values.size(); \
+      return Status(ONNXRUNTIME, FAIL, oss.str());                                                           \
```

Comment:
Should the error be INVALID_ARGUMENT?

Reply:
I changed it to "ORT_RETURN_IF". In this case, it is more like a logical programming error.